executor.go

```diff
@@ -278,15 +279,26 @@ func executeFields(p ExecuteFieldsParams) *Result {
 		p.Fields = map[string][]*ast.Field{}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(p.Fields))
+
+	var resultsMutex sync.Mutex
 	finalResults := map[string]interface{}{}
 	for responseName, fieldASTs := range p.Fields {
-		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs)
-		if state.hasNoFieldDefs {
-			continue
-		}
-		finalResults[responseName] = resolved
+		go func(responseName string, fieldASTs []*ast.Field) {
```

Comment:
Also, not sure if you know this, but goroutines are incredibly cheap and it's not uncommon to have thousands to goroutines at a time.

see https://talks.golang.org/2012/concurrency.slide#17